tests/test_time_frequency.py

```diff
@@ -377,5 +377,15 @@ def _test(layer, input_batch, allclose_func, atol=1e-4):
     )
 
 
+@pytest.mark.parametrize('hop_ratio', [0.5, 0.25, 0.125])
+def test_frequency_aware_conv2d(hop_ratio):
+    input_shape = (4, 28, 28, 3)
+    x = tf.random.normal(input_shape)
+    y = FreqAwareConv(
+     10, 3, activation='relu', input_shape=input_shape[1:])(x)
+    print(y.shape)
+    np.testing.assert_equal(y.shape, (4, 26, 26, 10))
+
+
```

Comment:
optional - feel free to add (or not) this layer to `test_save_load()` function which tests if it works well with model saving/loading and outputs the same!

Reply:
I have problems with the shape when embedding inside a ```tensorflow.keras.models.Sequential```. I didn't figure out the problem yet
```
import pytest
import numpy as np
import tensorflow as tf
import tensorflow.keras
import tensorflow.keras.backend as K
import librosa
from kapre.time_frequency import STFT, Magnitude, Phase, Delta, FrequencyAwareConv2D
from kapre.composed import (
    get_melspectrogram_layer,
    get_log_frequency_spectrogram_layer,
    get_stft_mag_phase,
    get_perfectly_reconstructing_stft_istft,
)
import tempfile

input_shape = (4, 50, 50, 1)
x = tf.random.normal(input_shape)
y = FrequencyAwareConv2D(10, 3, activation='relu', input_shape=input_shape[1:])(x)
np.testing.assert_equal(y.shape, (4, 48, 48, 10))
SRC = np.load('tests/speech_test_file.npz')['audio_data']

def get_audio(data_format, n_ch):
    src = SRC
    src = src[:16000]
    src_mono = src.copy()
    len_src = len(src)
    src = np.expand_dims(src, axis=1)  # (time, 1)
    if n_ch != 1:
        src = np.tile(src, [1, n_ch])  # (time, ch))
    if data_format == 'default':
        data_format = K.image_data_format()
    if data_format == 'channels_last':
        input_shape = (len_src, n_ch)
    else:
        src = np.transpose(src)  # (ch, time)
        input_shape = (n_ch, len_src)
    batch_src = np.expand_dims(src, axis=0)  # 3d batch input
    return src_mono, batch_src, input_shape



def _test(layer, input_batch, allclose_func, atol=1e-4):
    """test a model with `layer` with the given `input_batch`.
    The model prediction result is compared using `allclose_func` which may depend on the
    data type of the model output (e.g., float or complex).
    """
    model = tensorflow.keras.models.Sequential()
    model.add(layer)
    result_ref = model(input_batch)
    os_temp_dir = tempfile.gettempdir()
    model_temp_dir = tempfile.TemporaryDirectory(dir=os_temp_dir)
    model.save(filepath=model_temp_dir.name)
    print("saveeed")
    new_model = tf.keras.models.load_model(model_temp_dir.name)
    result_new = new_model(input_batch)
    allclose_func(result_ref, result_new, atol)
    model_temp_dir.cleanup()
    return model

src_mono, batch_src, input_shape = get_audio(data_format='channels_last', n_ch=1)

FrequencyAwareConv2D(256, 3, activation='relu',input_shape=(128,128,1
	))(get_melspectrogram_layer(input_shape=input_shape, return_decibel=True)(batch_src))

model = tensorflow.keras.models.Sequential()
model.add(FrequencyAwareConv2D(256, 3, activation='relu',input_shape=(128,128,2
	)))
model(get_melspectrogram_layer(input_shape=input_shape, return_decibel=True)(batch_src))


_test(    FrequencyAwareConv2D(256, 3, activation='relu'),
    get_melspectrogram_layer(input_shape=input_shape, return_decibel=True)(batch_src),
    np.testing.assert_allclose,
)


c=tensorflow.keras.models.Sequential(tensorflow.keras.models.Sequential([
    get_melspectrogram_layer(input_shape=input_shape, return_decibel=True),
    FrequencyAwareConv2D(256, 3, activation='relu',input_shape=(128,128,1))]))

# test FrequencyAwareConv2D
_test(tensorflow.keras.models.Sequential([
    get_melspectrogram_layer(input_shape=input_shape, return_decibel=True),
    FrequencyAwareConv2D(256, 3, activation='relu',input_shape=(128,128,1))]),
    batch_src,
    np.testing.assert_allclose,
)



```

Reply:
aha the non-batch input shape only needs to be specified at the first layer of `Sequential`. Seems like it's passed in other places too? For example, at FrequencyAwareConv2D after melspectorgram, you don't need to set it.